File: 1788/1788.go
func maximizeTheBeautyOfTheGarden(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maximizeTheBeautyOfTheGarden(root.Left)
	r := maximizeTheBeautyOfTheGarden(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 1788/1788.cs
public class Solution {
    public int MaximizeTheBeautyOfTheGarden(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MaximizeTheBeautyOfTheGarden(root.left), MaximizeTheBeautyOfTheGarden(root.right));
    }
}